tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionsProcessor.cs

```diff
@@ -41,7 +41,7 @@ internal void AddProbeProcessor(ProbeDefinition probe)
             {
                 _processors.AddOrUpdate(
                     probe.Id,
-                    new ProbeProcessor(probe),
+                    _ => new ProbeProcessor(probe),
```

Comment:
Is this adding an extra allocation?

Reply:
The opposite. Before this change we were allocating `ProbeProcessor` for no reason (same bug we had with `ProbeRateLimiter` leakage a while back).
My change is invoking the correct overload.